postgres/tests/test_statements.py

```diff
@@ -497,6 +500,220 @@ def test_statement_samples_collect(
+                    'xact_start': datetime.datetime(2021, 9, 22, 22, 21, 21, 669330, tzinfo=DEFAULT_TZ_INFO),
+                    'query_start': datetime.datetime(2021, 9, 19, 23, 21, 21, 669330, tzinfo=DEFAULT_TZ_INFO),
+                },
+                {
+                    'datname': 'datadog_test',
+                    'usename': 'veryoldbob',
+                    'xact_start': datetime.datetime(2021, 9, 20, 23, 21, 21, 669330, tzinfo=DEFAULT_TZ_INFO),
+                    'query_start': datetime.datetime(2021, 9, 20, 23, 21, 21, 669330, tzinfo=DEFAULT_TZ_INFO),
+                },
+            ]
+        ),
+        (
+            [
+                {
+                    'datname': 'datadog_test',
+                    'usename': 'newbob',
+                    'query_start': datetime.datetime(2021, 9, 23, 23, 21, 21, 669330, tzinfo=DEFAULT_TZ_INFO),
+                },
+                {
+                    'datname': 'datadog_test',
+                    'usename': 'oldbob',
+                    'xact_start': datetime.datetime(2021, 9, 22, 22, 21, 21, 669330, tzinfo=DEFAULT_TZ_INFO),
+                    'query_start': datetime.datetime(2021, 9, 22, 22, 21, 21, 669330, tzinfo=DEFAULT_TZ_INFO),
+                },
+                {
+                    'datname': 'datadog_test',
+                    'usename': 'veryoldbob',
+                    'query_start': datetime.datetime(2021, 9, 20, 23, 21, 21, 669330, tzinfo=DEFAULT_TZ_INFO),
+                },
+            ]
+        ),
+        (
+            [
+                {
+                    'datname': 'datadog_test',
+                    'usename': 'newbob',
+                    'query_start': datetime.datetime(2021, 9, 23, 23, 21, 21, 669330, tzinfo=DEFAULT_TZ_INFO),
+                },
+                {
+                    'datname': 'datadog_test',
+                    'usename': 'sameoldtxbob',
+                    'xact_start': datetime.datetime(2021, 9, 20, 23, 21, 21, 669330, tzinfo=DEFAULT_TZ_INFO),
+                    'query_start': datetime.datetime(2021, 9, 21, 23, 21, 21, 669330, tzinfo=DEFAULT_TZ_INFO),
+                },
+                {
+                    'datname': 'datadog_test',
+                    'usename': 'sameoldtxboblongquery',
+                    'xact_start': datetime.datetime(2021, 9, 20, 23, 21, 21, 669330, tzinfo=DEFAULT_TZ_INFO),
+                    'query_start': datetime.datetime(2021, 9, 20, 23, 21, 21, 669330, tzinfo=DEFAULT_TZ_INFO),
+                },
+            ]
+        ),
+    ],
+)
+def test_truncate_activity_rows(aggregator, integration_check, dbm_instance, active_rows):
+    # set max active rows to truncate to the two longest running transactions
+    dbm_instance['query_activity']['max_active_rows'] = 2
+    check = integration_check(dbm_instance)
+    check._connect()
+    with mock.patch(
+        'datadog_checks.postgres.statement_samples.PostgresStatementSamples._to_active_sessions',
```

Comment:
If we're mocking it anyway do we even need to query the actual database here? 

Seems we could simply call `_truncate_activity_rows`?

Reply:
oh yeah, derp that makes more sense